### ISSUE TITLE
err: sql: converting argument $1 type: unsupported type []uint64, a slice of uint64

### ISSUE THREAD

**Issue by 242617** [mattn/go-oci8#202]
I have an error
```
err: sql: converting argument $1 type: unsupported type []uint64, a slice of uint64
```
while passing uint64 array into db.QueryRow.

```
var IDs []uint64
...
if err = db.QueryRow(query, IDs).Scan(&result); err != nil {
	return
}
```

How should I pass such a param?

**Comment by pvelder** [comment]
Isn't the [variadic syntax](https://gobyexample.com/variadic-functions) working, using "..." after IDs ?

    var IDs []uint64
    ...
    if err = db.QueryRow(query, IDs...).Scan(&result); err != nil {
        return
    }

**Comment by 242617** [comment]
Unfortunately. I have query like this:
```
SELECT COLUMN_VALUE AS AGT_ID
FROM TABLE(IDS_TAB(:1))
MINUS
SELECT AGT_ID
FROM TBL_1 TBL
WHERE TBL.PRS_ID = :2
```
And []uint64 are for :1 param.

**Comment by pvelder** [comment]
OK, I see.

This is related to https://github.com/golang/go/issues/16235. 
See also commit "database/sql: allow drivers to support custom arg types" https://github.com/golang/go/commit/a9bf3b2e19920f6f516bc2e0211ae2e2f7ce395c 

@mattn don't see the interface driver.NamedValueChecker is implemented in go-oci8?

**Comment by mattn** [comment]
> don't see the interface driver.NamedValueChecker is implemented in go-oci8?

Yes, but always return true. Then, the slice will be converted (if I will implement converter).

**Comment by mattn** [comment]
This error comes from database/sql, not from this driver: driver arguments may only be primitive types and []byte, so a []uint64 (or any slice) cannot be passed as a single bind. For an IN list, expand the slice into individual placeholders ("in (:1, :2, :3)") and pass each element as its own argument, or insert the IDs into a temporary table and join against it.
